File: MissionRocketLaunch/MissionTeegardenB/app/src/main/java/com/flame4ost/missionteegardenb/Rocket.java
package com.flame4ost.missionteegardenb;

public class Rocket implements SpaceShip{

    int cost;
    int weight;
    int max_weight;
    double launch_explosion;
    double landing_crash;
    int current_weight;


    public boolean launch() {
        return true;
    }

    public boolean land() {
        return true;
    }

    public boolean canCarry(Item item) {
        return this.current_weight + item.weight <= max_weight;
    }

    public int carry(Item item) {
        this.current_weight += item.weight;
        return this.current_weight;
    }

    protected int getCost() {
        return cost;
    }
}

File: MissionRocketLaunch/MissionTeegardenBKotlin/app/src/main/java/com/flame4ost/missionteegardenbkotlin/Rocket.kt
package com.flame4ost.missionteegardenbkotlin

open class Rocket: SpaceShip {
    internal var cost: Int = 0
    var weight: Int = 0
    var max_weight: Int = 0
    var launch_explosion: Double = 0.0
    var landing_crash: Double =0.0
    var current_weight: Int = 0

    override fun launch(): Boolean { return true }

    override fun land(): Boolean{ return true }

    override fun canCarry(item: Item?): Boolean{
        return this.current_weight + item!!.weight <= max_weight}

    override fun carry(item: Item?): Int{
        this.current_weight +=item!!.weight;
        return this.current_weight
    }

    open fun getCost(): Int{ return this.cost}
}